### PR TITLE
Ensure the factors in referral program account for set eligibility

### LINKED ISSUE #9921: Referral program rewards set whereas referral set not eligible
Referral reward is computed whereas the referral set is not eligible. It should have been set to 0.

### PR DESCRIPTION
**Pull request by ValentinTrinque**
Close #9921

### CODE REVIEW

File: core/referral/engine.go [hunk 1]
```diff
@@ -484,95 +459,85 @@ func (e *Engine) computeReferralSetsStats(ctx context.Context, epoch types.Epoch
...
-			setStats.WasEligible = true
+		if setStats.WasEligible {
+			setStats.ReferralSetRunningVolume = e.referralSetsNotionalVolumes.RunningSetVolumeForWindow(setID, e.currentProgram.WindowLength)
+			setStats.RewardFactor = e.matchRewardFactor(setStats.ReferralSetRunningVolume)
+			setStats.RewardsMultiplier = e.matchRewardMultiplier(stakingBalance)
+			setStats.RewardsFactorMultiplier = num.MinD(
+				setStats.RewardFactor.Mul(setStats.RewardsMultiplier),
+				e.referralProgramMaxRewardProportion,
+			)
 		}
```
Comment (jeremyletang) [inline core/referral/engine.go:505]: I suppose we should still set the ReferralSetRunningVolume to something, even if not eligible, we'd like to know the running volume.

For the factor and multiplies I'd also set them to defaults values.

Comment (ValentinTrinque) [reply core/referral/engine.go:511]: > For the factor and multiplies I'd also set them to defaults values.

That is done, it's in the struct initializer.

File: core/referral/engine_test.go [hunk 2]
```diff
@@ -25,6 +25,7 @@ import (
 	"code.vegaprotocol.io/vega/libs/num"
 	vgrand "code.vegaprotocol.io/vega/libs/rand"
 	vgtest "code.vegaprotocol.io/vega/libs/test"
+
```
Comment (EVODelavega) [inline core/referral/engine_test.go:28]: RMBL

File: core/referral/engine.go [hunk 3]
```diff
@@ -138,6 +142,10 @@ func (e *Engine) CreateReferralSet(ctx context.Context, party types.PartyID, det
 		return ErrIsAlreadyAReferee(party)
 	}
 
+	if err := e.isPartyEligible(party); err != nil {
```
Comment (ze97286) [inline core/referral/engine.go:145]: I don't think this check is needed. We already have a spam check that won't allow a party with insufficient funds to createReferralSet.

**Review by EVODelavega — DISMISSED** [review]

**Review by EVODelavega — APPROVED** [review]

**Pull request merged**